frontend/src/components/dialogs/GameDetailsDialog.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import {Dialog, DialogContent, DialogTitle, Stack, Typography} from '@mui/material';
+import {useDialog} from './useDialog.ts';
+import {default as _ReactPlayer} from 'react-player/lazy';
+import {ReactPlayerProps} from "react-player/types/lib";
+
+const ReactPlayer = _ReactPlayer as unknown as React.FC<ReactPlayerProps>;
+
+export interface GameDetailsDialogProps {
+    title: string;
+    description: string;
+    releaseDate: string;
+    developerName: string;
+    score: string;
+    youtubeTrailer: string;
+}
+
+const GameDetailsDialog: React.FC = () => {
+    const {dialogProps, closeDialog} = useDialog();
+
+    if (!dialogProps || !(dialogProps as GameDetailsDialogProps).title) {
```

Comment:
Typecasts sind grundsätzlich erst mal immer gefährlich und sollte man versuchen zu vermeiden. Gibt dein useDialog dir nicht schon den GameDetailDialogProps type zurück? In dem Fall könntest du hier einfach `if(dialogProps?.title === null` (oder undefined, ist in deinem type aktuell nicht definiert. ;) )